Support for required options

woodruffw opened the pull request:
I've been needing this in one of my own projects and I saw that someone previously requested it in #203, so I went ahead and implemented it :smile: 

Let me know if you'd like me to update the README and/or Gem version in this PR as well.

Thanks for the awesome library!

Fixes #203

olleolleolle commented on lib/slop/parser.rb:
```diff
@@ -80,6 +80,13 @@ def parse(strings)
       @arguments += ignored_args
 
+      unused_options.each do |o|
+        if o.config[:required] && !suppress_errors?
```
Perhaps the whole loop can be obviated when `!suppress_errors?`

woodruffw replied:
Very true. I'll fix that.

olleolleolle commented:
This is a cool change. Can you teach the `required: true` option somehow, in some text file or the like?

woodruffw commented:
> Can you teach users the required: true option somehow, in some text file or the like?

Sure thing. Should I add it to the README, or is somewhere else better?

olleolleolle commented:
@woodruffw README's best!

olleolleolle commented:
(To fully answer: gem version should not be changed with a feature. That's a separate release acitivity.)

woodruffw commented:
Updated the README, plus I added `Option#required?` for consistency with `Option#suppress_errors?` and `Option#default_value`.

> (To fully answer: gem version should not be changed with a feature. That's a separate release acitivity.)

:+1:

woodruffw commented:
Any update on this @olleolleolle?

woodruffw commented:
Thanks! Would it be possible to get a version released for this change? I'd love to start using it :smile:

leejarvis commented:
Thanks both. I've released 4.6.0
